Add text about consistency and privacy reduction (#148)
ietf-wg-ohai/oblivious-http: draft-ietf-ohai-ohttp.md

requests using the same key configuration by matching KeyConfig.key\_id, or, if
the Target Resource is willing to use trial decryption, a limited set of key
configurations that share an identifier. An Oblivious Relay Resource can link
requests using the public key corresponding to KeyConfig.key\_id or via the

> Oblivious Gateway Resource. We refer to this set of information as the gateway's

Reviewer: ```suggestion
identity of the Oblivious Gateway Resource. We refer to this set of information as the gateway's
```

Reviewer: Or
```suggestion
URI of the Oblivious Gateway Resource. We refer to this set of information as the gateway's
```

Author: Right, I just left identity and URI out so it generally refers to "the resource." I'm OK with URI though.

configuration.

Whether or not targets can link requests depends on how gateway configuration
information is produced and discovered by clients. Specifically, gateways can
maliciously construct configurations to track individual clients.

Ideally, all clients share a consistent view of the gateway configuration. The number
of different valid configurations can exponentially partition the anonymity set of
clients. For example, if there are two valid configurations, that would yield two

> anonymity sets consisting of clients that used one configuration and clients that used
> the other configuration. If the size of these sets are large enough and roughly equal,
> this may not be a total loss of privacy in practice. But it is possible that one
> set is small while the other is large. Configuration consistency is an important
> part of ensuring client anonymity sets are few in number and large in size.

Reviewer: ```suggestion
anonymity sets consisting of clients that used one configuration and clients that used
the other configuration. If the size of each set is large enough, this may not be
a significant loss of privacy in practice. Ensuring that many clients share
a configuration is necessary to provide privacy for clients.
```

A specific method for a client to acquire configurations is not included in this

> specification. Clients need to consider these tracking vectors when choosing a discovery
> method. Applications using this design should provide accommodations to mitigate tracking

Reviewer: ```suggestion
specification. Applications using this design MUST provide accommodations to mitigate tracking
```

> using gateway configurations. {{?CONSISTENCY=I-D.wood-key-consistency}} provides an
> analysis of the options for ensuring the configurations are consistent between
> different clients.

Reviewer: ```suggestion
using gateway configurations. {{?CONSISTENCY=I-D.wood-key-consistency}} provides
options for ensuring that configurations are consistent between clients.
```
